prody/proteins/interactions.py

```diff
@@ -0,0 +1,2043 @@
+            sele2_nr = list(set(zip(sele2.getResnums(), sele2.getChids())))
+
+            if sele1_name[0] in aromatic:
+                sele2_filter = sele2.select('all and not (resname TYR PHE TRP or resid '+str(i)+')')
+                if sele2_filter != None:
+                    listOfAtomToCompare = cleanNumbers(findNeighbors(sele1, distA, sele2_filter))
+                
+            elif sele1_name[0] not in aromatic and i in sele2_nr:
+                sele2_filter = sele2.select(sele2.select('all and not (resid '+str(i[0])+' and chain '+i[1]+')'))
+                if sele2_filter != None:
+                    listOfAtomToCompare = cleanNumbers(findNeighbors(sele1, distA, sele2_filter))
+            else:
+                listOfAtomToCompare = cleanNumbers(findNeighbors(sele1, distA, sele2))
+                                                           
+            if listOfAtomToCompare != []:
+                listOfAtomToCompare = sorted(listOfAtomToCompare, key=lambda x : x[-1])
+                minDistancePair = listOfAtomToCompare[0]
+                if minDistancePair[-1] < distA:
+                    sele1_new = atoms.select('index '+str(minDistancePair[0])+' and name '+str(minDistancePair[2]))
+                    sele2_new = atoms.select('index '+str(minDistancePair[1])+' and name '+str(minDistancePair[3]))
+                    Hydrophobic_calculations.append([sele1_new.getResnames()[0]+str(sele1_new.getResnums()[0]), 
+                                                             minDistancePair[2]+'_'+str(minDistancePair[0]), sele1_new.getChids()[0],
+                                                             sele2_new.getResnames()[0]+str(sele2_new.getResnums()[0]), 
+                                                             minDistancePair[3]+'_'+str(minDistancePair[1]), sele2_new.getChids()[0],
+                                                             round(minDistancePair[-1],3)]) 
+                    
+    Hydrophobic_calculations = sorted(Hydrophobic_calculations, key=lambda x : x[-1])
+    Hydrophobic_calculations_final = removeDuplicates(Hydrophobic_calculations)
+    
+    for kk in Hydrophobic_calculations_final:
+        LOGGER.info("%10s%5s%14s  <---> %10s%5s%14s%8.1f" % (kk[0], kk[2], kk[1], kk[3], kk[5], kk[4], kk[6]))
+        
+    LOGGER.info("Number of detected hydrophobic interactions: {0}.".format(len(Hydrophobic_calculations_final)))
+    
+    return Hydrophobic_calculations_final
+
+
+def calcMetalInteractions(atoms, distA=3.0, extraIons=['FE'], excluded_ions=['SOD', 'CLA']):
+    """Interactions with metal ions (includes water, ligands and other ions).
+        
+    :arg atoms: an Atomic object from which residues are selected
+    :type atoms: :class:`.Atomic`
+    
+    :arg distA: non-zero value, maximal distance between ion and residue.
+    :type distA: int or float, default is 3.0.
+    
+    :arg extraIons: ions to be included in the analysis.
+    :type extraIons: list of str
```

Comment:
list of str should be commented in the main description and only list should be stated as the type